include/gevolution/lightcone.hpp

```diff
@@ -0,0 +1,287 @@
+#pragma once
+
+#include <gevolution/basic_types.hpp>
+
+#include <boost/mpi/communicator.hpp>
+#include <boost/mpi/collectives.hpp>
+#include <boost/serialization/vector.hpp>
+#include <boost/serialization/utility.hpp>
+
+#include <cmath>
+#include <set>
+#include <map>
+#include <vector>
+#include <utility>
+
+#ifdef HAVE_HEALPIX
+#include "chealpix.h"
+#ifndef PIXBUFFER
+#define PIXBUFFER 1048576
+#endif
+#endif
+
+namespace gevolution {
+
+
+#ifdef HAVE_HEALPIX
+struct healpix_header
+{
+    uint32_t Nside;
+    uint32_t Npix;
+    uint32_t precision;
+    uint32_t Ngrid;
+    double direction[3];
+    double distance;
+    double boxsize;
+    uint32_t Nside_ring;
+    char fill[256 - 5 * 4 - 5 * 8]; /* fills to 256 Bytes */
+};
+#endif
+
+struct lightcone_geometry
+{
+    // comoving position of the observer 
+    double vertex[3];
+    
+    // redshift of the observation event
+    double z;
+    
+    // conformal time of the observation event 
+    // double tau;
+    
+    // direction of the axis of the lightcone
+    double direction[3];
+    
+    // cosine of the angular opening of the geometric lightcone
+    double opening;
+    
+    // comoving radius of the lightcone shell
+    // distance[0] < distance[1]
+    double distance[2]; 
+    
+    template<class Archive>
+    void serialize(Archive & ar, const unsigned int /*version*/)
+    // Boost serialization of this class.
+    // needed in order to broadcast this data using MPI
+    {
+        ar & vertex;
```

Comment:
I am not sure that this will work with raw C arrays.

Reply:
Yes, boost serialization handles arrays: https://www.boost.org/doc/libs/1_80_0/libs/serialization/doc/tutorial.html#arrays